### PR TITLE
build: configure cross-pkg resolution for api extraction

### PR DESCRIPTION
**Pull request by jelbourn**
This commit adds path mapping and source dependencies necessary to fully resolve types during api doc extraction.

### CODE REVIEW

**Review by josephperrott — APPROVED** [review]
LGTM

File: tools/esm-interop/patches/npm/@angular+build-tooling+0.0.0-e0ec7b60641d7f6369be45d8d02663fd50f320be.patch [hunk 1]
```diff
@@ -28,17 +28,19 @@ index afe4e5c..095a8f0 100755
...
      data = [
          ":bin",
 -        "@npm//@angular/compiler",
 -        "@npm//@angular/compiler-cli",
 +        "@angular//packages/compiler",
 +        "@angular//packages/compiler-cli",
+         "@npm//typescript",
      ],
      entry_point = "bin.mjs",
+<<<<<<<< HEAD:tools/esm-interop/patches/npm/@angular+build-tooling+0.0.0-b2cec2dcba358f9dc7111800a3d65738f57abe8f.patch
```
Comment (AndrewKushnir) [inline tools/esm-interop/patches/npm/@angular+build-tooling+0.0.0-e0ec7b60641d7f6369be45d8d02663fd50f320be.patch:43]: It looks like this is an artifact of the merge conflict, should we revert it?

**Comment by jelbourn** [comment]
@AndrewKushnir came back to fix this up after the holidays, should be good to go now

**Review by AndrewKushnir — APPROVED** [review]

**Comment by jelbourn** [comment]
Current set of approvals should be good since this is all build/docs changes rather than framework code

**Review by atscott — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit 91f250dab7dcd6aacab04cb4cfff321aa7b6001f.

**Pull request closed without merging**